timed/src/statistics.rs

```diff
@@ -22,15 +22,10 @@ impl StatisticsRecord {
         }
     }
 
-    pub fn nth_percentile_time(&self, percentile: f32) -> Option<&Duration> {
-        let mut calls = self.calls.clone();
-        calls.sort();
```

Comment:
Problem was that I cloned the `calls` vec, sorted it, and then called `self.calls.get(i)` instead of just `calls.get(i)` (on the cloned vec).

I decided to fix it by removing the call for clone, changing the signature to receive a mutable reference to `self` and sort the `calls` vec in place (on `self`).
This way its more readable, less error prune and less `memcp`s.
Wdyt?